src/viewer/Viewer.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+
+import ViewerToolbar from './ViewerToolbar'
+import ImageViewer from './ImageViewer'
+import AudioViewer from './AudioViewer'
+import VideoViewer from './VideoViewer'
+import NoViewer from './NoViewer'
+
+import styles from './viewer'
+
+const KEY_CODE_LEFT = 37
+const KEY_CODE_RIGHT = 39
+
+export default class Viewer extends Component {
+  state = {
+    controlsHidden: false
+  }
+
+  showControls = () => {
+    this.setState({ controlsHidden: false })
+    document.removeEventListener('mousemove', this.showControls)
+  }
+  hideControls = () => {
+    this.setState({ controlsHidden: true })
+    document.addEventListener('mousemove', this.showControls)
+  }
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.onKeyUp, false)
```

Comment:
`keyup` is a better event to catch a `onNext` behavior. But we cannot fast browse. Maybe we could add an event listener to accept fast browsing with `keydown` pressed for at least 1000 ms and that stops on the image when keydown is released... (I can explain it in french if I am not clear.)

Reply:
I see what you mean, but I'm afraid it'd take a long time to implement properly (just a few hours probably, but I'd like to validate this story tonight), so... next time? :(